AppCenter/AppCenter/Internals/Device/MSDeviceTracker.m

```diff
@@ -111,9 +111,40 @@ - (MSDevice *)device {
  */
 - (MSDevice *)updatedDevice {
   @synchronized(self) {
-    MSDevice *newDevice = [[MSDevice alloc] init];
+    MSDevice *newDevice = [MSDevice new];
 #if TARGET_OS_IOS
-    CTCarrier *carrier = [[[CTTelephonyNetworkInfo alloc] init] subscriberCellularProvider];
+    CTTelephonyNetworkInfo *telephonyNetworkInfo = [CTTelephonyNetworkInfo new];
+    CTCarrier *carrier;
+
+// TODO Use @available API and availability attribute when deprecating Xcode 8 then we can try removing these pragma.
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wundeclared-selector"
+   SEL serviceSubscriberCellularProviders = @selector(serviceSubscriberCellularProviders);
```

Comment:
You can use `NSSelectorFromString` to avoid suppress this warning.

Reply:
@jaelim-ms no we already tried: 0782403 and 4c61a12